src/expr.rs

```diff
@@ -1776,27 +1776,25 @@ fn rewrite_call_args(context: &RewriteContext,
         acc + item.item.as_ref().map_or(0, |s| 2 + first_line_width(s))
     }) <= one_line_budget + 2;
 
-    match write_list(&item_vec, &fmt) {
+    let result = write_list(&item_vec, &fmt);
+    let last_char = result
+        .as_ref()
+        .map_or(' ', |r| r.chars().last().unwrap_or(' '));
```

Comment:
rather than keeping the last char and (I think) using `' '` as a null value, you could keep a `last_char_is_comma` bool here.